docs/proposals/202012_receive_split.md

```diff
@@ -8,30 +8,79 @@ status: accepted
 ### Related Tickets
 
 * Hashring improvements: https://github.com/thanos-io/thanos/issues/3141
+* Previous proposal implementations: https://github.com/thanos-io/thanos/pull/3845, https://github.com/thanos-io/thanos/pull/3580
+* Distribution idea: https://github.com/thanos-io/thanos/pull/2675
 
 ### Summary
 
-This document describes the motivation and design of splitting the receive component into receive-route and receive.
-Additionally, we touch possibility for adding a consistent hashing mechanism and buffering the incoming requests.
+This document describes the motivation and design of running Receive in a stateless mode that does not have capabilities to store samples, it only routes remote write
+to further receives based on hashring. This allows setting optional deployment model were only Routing Receives are using hashring files and does the routing and replication. That allows ingesting Receivers to not handle any routing or hashring, only receiving multi tenant writes.
 
 ### Motivation
 
-1. Splitting functionality can be used to make resharding events easier.
-2. Right now the receiver handles ingestion, routing and writing, thus leading to too many responsibilities in a single process.
-This also makes the receiver more difficult to operate and understand for Thanos users.
-Splitting the receiver component into two different components could potentially have the following benefits:
-    1. Resharding events become faster and cause no downtime in ingestion.
-    2. Deployment becomes easier to understand for Thanos users.
-    3. Each component consists of less code.
-    4. The new architecture enables further performance improvements.
+[@squat](https://github.com/squat):
+
+> Currently, any change to the hashring configuration file will trigger all Thanos Receive nodes to flush their multi-TSDBs, causing them to enter an unready state until the flush is complete. This unavailability during a flush allows for a clear state transition, however it can result in downtimes on the order of five minutes for every configuration change. Moreover, during configuration changes, the hashring goes through an even longer period of partial unreadiness, where some nodes begin and finish flushing before and after others. During this partial unreadiness, the hashring can expect high internal request failure rates, which cause clients to retry their requests, resulting in even higher load. Therefore, when the hashring configuration is changed due to automatic horizontal scaling of a set of Thanos Receivers, the system can expect higher than normal resource utilization, which can create a positive feedback loop that continuously scales the hashring.
+
 
 ### Goals
 
+* Reduce the downtime of ingestion part of Thanos
```

Comment:
```suggestion
* Reduce downtime of the ingestion operation of Thanos Receiver
```

Reply:
Hm, wonder if operation is a right word 🤔

Reply:
changed to `logic`